### PR TITLE
feat: Align header with HMPPS Digital Services

### PR DESCRIPTION
**Pull request by Thource**
### What changed

The styling of the header has changed to match the HMPPS Digital Services header.

### Why did it change

This changed because it's nicer if everything matches style, also because there was no way to navigate back to HMPPS Digital Services from BaSM.

### Issue tracking

- [P4-2626](https://dsdmoj.atlassian.net/browse/P4-2626)

## Screenshots

| Before | After |
| ------ | ----- |
| <kbd>image</kbd> | <kbd>image</kbd> |

#### Automated testing

- [ ] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [ ] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [ ] No environment variables were added or changed

- [ ] Documented in the [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/main/README.md)
- [ ] Added to [continous integration](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-book-secure-move-frontend)
- [ ] Added to staging environment (deployment repository)
- [ ] Added to preproduction environment (deployment repository)
- [ ] Added to production environment (deployment repository)

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/main/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [ ] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [ ] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [inline common/components/internal-header/_internal-header.scss:1]: Could we revert these quote changes please. I suspect it might be an IDE setting you have. Although we don't have some form of scss linting applied yet the standard approach up until now has been double quotes so it would be good to stick to that and reduce the noise in this PR.

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [inline common/components/internal-header/_internal-header.scss:17]: Are we removing support for this modifier with these changes? Might be nice to see if we can still support it.

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [inline common/components/internal-header/_internal-header.scss:93]: Do we need this change? The value of `$govuk-header-link` defined on line 5 is already white. I think it would be nice to continue to support the configuration approach.

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [inline common/components/internal-header/_internal-header.scss:102]: All of these definitions are actually being applied by using `@include govuk-focused-text;` so maybe let's revert back to that so that if that changes in the future we'll benefit from it.

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [inline common/components/internal-header/_internal-header.scss:109]: Could we use this (from line 43):
```suggestion
      fill: currentColor;
```

This will mean that it will benefit from whatever the color defined is rather than having to be set separately.

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [inline common/components/internal-header/_internal-header.scss:175]: Are we able to clean up the use of hardcoded colours from these definitions. Maybe we can use one of the variables defined at the start of this file?

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [inline common/components/internal-header/_internal-header.scss:292]: These statements might be worth keeping so that the links for "sign out" aren't printed for example

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [inline common/components/internal-header/_internal-header.scss:299]: I think it would also be good to keep these at the bottom of the file as overrides. They're things currently required to work with the font we have and these allow those fixes to stand out quite nicely.

File: common/components/internal-header/template.njk
Comment (teneightfive) [inline common/components/internal-header/template.njk:3]: You can shorten this to:
```suggestion
  <div class="app-header__container {{ params.containerClasses }}">
```

If it has no value nunjucks will not output anything.

File: common/components/internal-header/template.njk
Comment (teneightfive) [inline common/components/internal-header/template.njk:6]: I'm a bit conscious of having this variables within the component as now they are tied specifically to the app configuration and are no longer independent. 

I think a nice approach would be to update the value of `homepageUrl` to `AUTH_BASE_URL + "/auth"` in [base.njk](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/d3b9972af9a0c322de782586255d9a57c05adc06/common/templates/layouts/base.njk#L96) where we define the header. Then the value is tied specifically to our implementation of this component and when testing the component you don't need to worry about setting global nunjucks values (with I think might be causing some of the unit test failures).

File: common/components/internal-header/template.njk
Comment (teneightfive) [inline common/components/internal-header/template.njk:60]: I think this should be:
```suggestion
                  <span class="app-header__link-sub-text">
```

to better fit with the [BEM](http://getbem.com/) naming convention.

File: common/components/internal-header/template.test.js
Comment (teneightfive) [inline common/components/internal-header/template.test.js:1]: I've left a comment on the template file as to how we can remove the need to include our app configuration from this component. I think it would be a much cleaner separation to keep the two separate.

File: config/nunjucks/globals.js
Comment (teneightfive) [inline config/nunjucks/globals.js:18]: This file is already included on line 3 so you can add a definition to that instead of including it again.
```suggestion
```

File: test/e2e/#smoke.test.js
Comment (teneightfive) [inline test/e2e/#smoke.test.js:66]: This could be changed to `.contains` so you don't need to worry about it including the manage link.

**Review by teneightfive — CHANGES_REQUESTED** [review]
I've left quite a few comments around the big changes to the styles and layout. Let me know if it's easier to go through them together.

I've also looked at the changes using different breakpoints and the mobile view seems to be missing some padding that I think we should address:

<kbd>image</kbd>

compared to:

<kbd>image</kbd>

I've also got a slight concern that with the changes to smaller screen widths there is now no way to access the homepage for our service as we've removed the service name. Previously it stacked the service name and I wonder if we should continue that:

<kbd>image</kbd>

@elizabethgoddard what do you think?

Another wider thing @elizabethgoddard and I talked about it making the container size increase with this same change. Currently this would make the header now wider than our content which we felt looked a little odd:

<kbd>image</kbd>

So instead, we could look to override the value for `$govuk-page-width` and `$moj-page-width` in our [application.scss](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/d3b9972af9a0c322de782586255d9a57c05adc06/common/assets/scss/application.scss#L6) file. Which will produce something more like this:

<kbd>image</kbd>

File: common/components/internal-header/_internal-header.scss
Comment (Thource) [reply common/components/internal-header/_internal-header.scss:17]: From what I could see, it wasn't used anywhere in the project, the styles that it added are included in `.app-header__container` now (except for border-color, which had no effect outside of `container--with-border`, which also wasn't used anywhere)

File: common/components/internal-header/_internal-header.scss
Comment (Thource) [reply common/components/internal-header/_internal-header.scss:93]: I knew I'd have missed one, I copied the styles from hmpps-auth and converted them to scss

File: common/components/internal-header/_internal-header.scss
Comment (Thource) [reply common/components/internal-header/_internal-header.scss:292]: Hmm, the whole header is being hidden during print now (same behaviour as HMPPS DS), do we want the header to be visible?

File: common/components/internal-header/_internal-header.scss
Comment (Thource) [reply common/components/internal-header/_internal-header.scss:299]: I believe I may have fixed the issue that required these styles to be necessary? I changed the SVG path and attrs to be the same as HMPPS DS and from what I can see, the text in the header and the crest are lined up perfectly? (same as HMPPS DS)

**Comment by ElizabethGoddard** [comment]
Hey @Thource and @teneightfive, sorry for the delay was just chatting to Dom and also separately with DPS.

> I've also looked at the changes using different breakpoints and the mobile view seems to be missing some padding that I think we should address

Yes please to fixing the padding :)

> I've also got a slight concern that with the changes to smaller screen widths there is now no way to access the homepage for our service as we've removed the service name. Previously it stacked the service name and I wonder if we should continue that

Had a chat with Dom - I'm not too concerned because we have access via 'Home' in the primary navigation, however police users do not currently. We will probably give police users the home dashboard at some point for consistency though. Hold this thought because I'm just checking with DPS what their other services do before we decide - they are going to get back to me.

> Another wider thing @ElizabethGoddard and I talked about it making the container size increase with this same change. Currently this would make the header now wider than our content which we felt looked a little odd

@Thource, Dom mentioned you looked at DPS and they have a 960 container. Actually this is only on the landing page (they tell me this is owned by someone else but they are looking to update it to the wider grid). If you click into DPS or other services you'll see it actually is the wider grid (1170px). If we could update to match that would be preferable because that landing page will eventually match and we will then also match DPS and associated services. Also like Dom says it looks more balanced with the header and contents in alignment.

Thanks guys!

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [reply common/components/internal-header/_internal-header.scss:1]: Ah right, I see. I think the intention of that definition is for `.js` files though as that's also a rule we use for JavaScript linting.

Currently almost all of our `.scss` files use double quotes. I might look to clarify this with either adding an extension override to the `.prettierrc` or adding style linting. 

But if we could revert it as part of this change for now that would keep it consistent with the rest of the files of the same type.

File: common/components/internal-header/template.njk
Comment (Thource) [reply common/components/internal-header/template.njk]: Replaced `AUTH_BASE_URL + "/auth"` with `organisationHomepageUrl`, homepageUrl is already in use for the BaSM homepage

File: .prettierrc
Comment (teneightfive) [inline .prettierrc:8]: 👍

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [inline common/components/internal-header/_internal-header.scss:106]: This looks like its nested within the `@include govuk-compatibility`, is that GitHub diffs playing up or intentional?

I'm guessing we might not need it nested if so?

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [reply common/components/internal-header/_internal-header.scss:292]: Yes, that's something we're doing at an app level but I was thinking more at a component level for these styles. If its ever extracted and put into a service where the header isn't removed I think it would be useful to keep these styles to prevent the links being shown.

But maybe that's extra page weight...I'll leave that as your call

File: common/components/internal-header/_internal-header.scss
Comment (teneightfive) [reply common/components/internal-header/_internal-header.scss:299]: Yes, that's true but it was more the approach if settings the adjustments at the end of the page. I think the same minus top position has just been moved inside that statement instead of noted as an adjustment.

It feels like we've strayed enough to maybe not warrant needing them defined like this so fine to leave it. The adjustments were to cater for not using GDS Transport (I think).

File: test/e2e/#smoke.test.js
Comment (teneightfive) [inline test/e2e/#smoke.test.js:15]: Do we still need both `username` and `displayName`? I feel like it was only used for the smoke tests that you've changed to use `displayName`?

**Review by teneightfive — APPROVED** [review]
Couple minor comments but looks good to me 👍

File: common/components/internal-header/_internal-header.scss
Comment (Thource) [reply common/components/internal-header/_internal-header.scss:106]: That's just GitHub diffs playing up, this is nested as `.app-header__link { &:focus { svg {`

**Pull request merged**
